src/style.rs

```diff
@@ -771,6 +769,38 @@ impl fmt::Display for PaddedStringDisplay<'_> {
     }
 }
 
+/// Write the visible text between start and end. The ansi escape
+/// sequences are written unchanged.
+pub fn write_ansi_range(
+    formatter: &mut Formatter,
+    text: &str,
+    start: usize,
+    end: usize,
+) -> fmt::Result {
+    let mut pos = 0;
+    for (s, is_ansi) in AnsiCodeIterator::new(text) {
+        if is_ansi {
+            formatter.write_str(s)?;
+        } else if pos < end {
```

Comment:
```rust
} else if pos >= end {
    break;
}
```

?

Reply:
Do you think we can skip the rest of the ansi escape sequences? It seems safer to keep them all